### PR TITLE
NVSHAS-9501 support SLSA

### PR DESCRIPTION
**Pull request by holyspectral**
Implement standalone Dockerfile and corresponding GHA to support SLSA L3.

### CODE REVIEW

File: package/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,21 @@
+FROM registry.suse.com/bci/python:3.12
+MAINTAINER support@neuvector.com
+
+RUN python3 -m venv .venv && source .venv/bin/activate && pip3 install --upgrade pip setuptools prometheus_client requests
+COPY startup.sh /usr/local/bin
+COPY nv_exporter.py /usr/local/bin
+
```
Comment (pjbgf) [inline package/Dockerfile:7]:
```suggestion

ARG VERSION
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+test-image:
+	# Instead of loading image, target all platforms, effectivelly testing
+	# the build for the target architectures.
+	$(MAKE) build-image BUILD_ACTION="--platform=$(TARGET_PLATFORMS)"
+
+build-image: buildx-machine ## build (and load) the container image targeting the current platform.
+	$(IMAGE_BUILDER) build -f package/Dockerfile \
+		--builder $(MACHINE) $(IMAGE_ARGS) \
+		--build-arg VERSION=$(VERSION) -t "$(IMAGE)" $(BUILD_ACTION) .
```
Comment (pjbgf) [inline Makefile:59]:
```suggestion
		--build-arg VERSION=$(VERSION) --build-arg COMMIT=$(COMMIT) -t "$(IMAGE)" $(BUILD_ACTION) .
```

**Review by pjbgf — APPROVED** [review]

**Review by pjbgf — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	$(IMAGE_BUILDER) build -f package/Dockerfile \
+		--builder $(MACHINE) $(IMAGE_ARGS) $(IID_FILE_FLAG) $(BUILDX_ARGS) \
+		--build-arg VERSION=$(VERSION) --build-arg COMMIT=$(COMMIT) --platform=$(TARGET_PLATFORMS) -t "$(REPO)/prometheus-exporter:$(TAG)" --push .
+	@echo "Pushed $(IMAGE)"
+
+push-rancher-image: buildx-machine
+	$(IMAGE_BUILDER) build -f package/Dockerfile \
+		--builder $(MACHINE) $(IMAGE_ARGS) $(IID_FILE_FLAG) $(BUILDX_ARGS) \
+		--build-arg VERSION=$(VERSION) --build-arg COMMIT=$(COMMIT) --platform=$(TARGET_PLATFORMS) -t "$(REPO)/neuvector-prometheus-exporter:$(TAG)" --push .
+	@echo "Pushed $(IMAGE)"
```
Comment (pjbgf) [inline Makefile:73]: No need to have an additional target, if you rely on `IMAGE`:
```suggestion
push-image: buildx-machine
	$(IMAGE_BUILDER) build -f package/Dockerfile \
		--builder $(MACHINE) $(IMAGE_ARGS) $(IID_FILE_FLAG) $(BUILDX_ARGS) \
		--build-arg VERSION=$(VERSION) --build-arg COMMIT=$(COMMIT) --platform=$(TARGET_PLATFORMS) -t "$(IMAGE)" --push .
	@echo "Pushed $(IMAGE)"
```

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+	TAG = $(VERSION)
+	ifneq ($(DIRTY),)
+		TAG = dev
+	endif
+endif
+
+TARGET_PLATFORMS ?= linux/amd64,linux/arm64
+STAGE_DIR=stage
+REPO ?= neuvector
+IMAGE = $(REPO)/prometheus-exporter:$(TAG)
```
Comment (pjbgf) [inline Makefile:41]:
```suggestion
IMAGE = $(REPO)/$(IMAGE-SUFFIX)prometheus-exporter:$(TAG)
```

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        public-registry: docker.io
+        public-repo: neuvector
+        public-username: ${{ env.DOCKER_USERNAME }}
+        public-password: ${{ env.DOCKER_PASSWORD }}
+    - name: Publish prime manifest
+      uses: rancher/ecm-distro-tools/actions/publish-image@master
+      with:
+        push-to-public: false
+        push-to-prime: true
+        make-target: push-rancher-image
```
Comment (pjbgf) [inline .github/workflows/release.yml:55]:
```suggestion
        make-target: push-image
```

File: .github/workflows/release.yml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        image: neuvector-prometheus-exporter
+        tag: ${{ env.TAG }}
+        platforms: linux/amd64,linux/arm64
+
+        prime-registry: ${{ env.PRIME_REGISTRY }}
+        prime-repo: rancher
+        prime-username: ${{ env.PRIME_REGISTRY_USERNAME }}
+        prime-password: ${{ env.PRIME_REGISTRY_PASSWORD }}
```
Comment (pjbgf) [inline .github/workflows/release.yml:63]:
```suggestion
        prime-password: ${{ env.PRIME_REGISTRY_PASSWORD }}
      env:
        IMAGE-SUFFIX: rancher-
```

**Review by pjbgf — APPROVED** [review]

**Review by BinX-Suse — APPROVED** [review]

**Pull request merged**